presto-common/src/main/java/com/facebook/presto/common/Page.java

```diff
@@ -456,6 +456,24 @@ private long updateRetainedSize()
         return retainedSizeInBytes.longValue();
     }
 
+    /**
+     * Returns a new page with the same columns as the original page except for the one column replaced.
+     *
+     * @param channelIndex the column to replace
+     * @param column the replacement column
+     * @return a new page with the replacement column substituted for the old column
+     */
+    public Page replaceColumn(int channelIndex, Block column)
+    {
```

Comment:
Check and throw if the channelIndex is invalid

Reply:
There are unit tests that verify the correct exception is thrown in this case.